### PR TITLE
FINAL CHANGES

### PR DESCRIPTION
**Pull request by waynepHMCTS**
Resolves # . (This is applicable only if this pull request relates to a GitHub issue, delete the line otherwise)

Notes:
*
*
*

### CODE REVIEW

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@waynepHMCTS - a couple of things to look at:
* Please remove `package-lock.json`. This has been superseded by `yarn.lock`
* Please add some unit tests for the new `http-global-interceptor`

**Review by kuda1992 — CHANGES_REQUESTED** [review]
You need to change the package.json version name from 1.4.3-display-case-filters" to 1.4.3-not-all-exceptions" or you won't be able to create a tag.

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@waynepHMCTS - still needs some unit tests for the new interceptor.

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@waynepHMCTS - thanks for adding a unit test. Is it possible to extend it to cover both branches of execution? I notice in the implementation there's a check for whether it's a client-side error or a server-side error.

**Review by udaydenduluri33 — APPROVED** [review]

**Review by kuda1992 — APPROVED** [review]

File: projects/exui-common-lib/src/lib/services/http-global-interceptors/http-global-interceptor.spec.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    describe('Error test', () => {
+      it('should call assignErrorMessage with the error raised', () => {
+        spyOn(errorInterceptor, 'assignErrorMessage');
+        const mock404Error = {
+          status: 404,
+          error: { message: 'not found' }
+        };
+
+        const next = {
+          handle: () =>  throwError(mock404Error)
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/http-global-interceptors/http-global-interceptor.spec.ts:29]: @waynepHMCTS - linting: extra whitespace after `=>`

File: projects/exui-common-lib/src/lib/services/http-global-interceptors/http-global-interceptor.spec.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        };
+
+        const handleError$ = errorInterceptor.intercept(new HttpRequest('GET', ''), next);
+
+        handleError$.subscribe((res) => {
+            console.log(res);
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/http-global-interceptors/http-global-interceptor.spec.ts:35]: @waynepHMCTS - linting: indentation (should be 2 spaces)

File: projects/exui-common-lib/src/lib/services/http-global-interceptors/http-global-interceptor.spec.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        }, () => {
+            expect(errorInterceptor.assignErrorMessage).toHaveBeenCalledWith(mock404Error);
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/src/lib/services/http-global-interceptors/http-global-interceptor.spec.ts:37]: @waynepHMCTS - linting: indentation (should be 2 spaces)

File: projects/exui-common-lib/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@hmcts/rpx-xui-common-lib",
-  "version": "1.4.3-enable-find-person",
+  "version": "1.4.3-add-interceptor-for-global-error-handling",
```
Comment (Daniel-Lam) [inline projects/exui-common-lib/package.json:3]: @waynepHMCTS - linting: extra whitespace at line end

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@waynepHMCTS - all looks good, except for a few, minor linting issues

**Review by Daniel-Lam — APPROVED** [review]

**Pull request merged**